rta/__init__.py

```diff
@@ -4,41 +4,77 @@
 # 2.0.
 
 import importlib
+import inspect
+from dataclasses import dataclass, field
 from pathlib import Path
-from typing import List, Optional
+from typing import Dict, List, Optional
 
 from . import common
 
 CURRENT_DIR = Path(__file__).resolve().parent
 
 
-def get_ttp_list(os_types: Optional[List[str]] = None) -> List[str]:
-    scripts = []
-    if os_types and not isinstance(os_types, (list, tuple)):
-        os_types = [os_types]
+@dataclass
+class RtaMetadata:
+    """Metadata associated with all RTAs."""
 
-    for script in CURRENT_DIR.glob("*.py"):
-        base_name = script.stem
-        if base_name not in ("common", "main") and not base_name.startswith("_"):
-            if os_types:
-                # Import it and skip it if it's not supported
-                importlib.import_module(__name__ + "." + base_name)
-                if not any(base_name in common.OS_MAPPING[os_type] for os_type in os_types):
-                    continue
+    uuid: str
+    platforms: List[str]
 
-            scripts.append(str(script))
+    path: Path = field(init=False)
+    name: str = field(init=False)
+    endpoint: Optional[List[dict]] = None
+    siem: Optional[List[dict]] = None
+    techniques: Optional[List[str]] = None
 
-    return scripts
+    def __post_init__(self):
+        """Set the path and name based on the callee."""
+        for frame in inspect.stack():
+            self.path = Path(frame.filename)
+            self.name = self.path.name
+            if frame.function == "<module>" and valid_rta_file(self.path):
+                break
```

Comment:
will this work with `Path(__file__)` instead?

Reply:
It may be possible, but when I tried it, it always used the class definition path instead of the instance.